File: ghidra/scripts/DumpReferences.py
# Print references to selected addresses with containing function and instruction.
# Usage: -postScript DumpReferences.py 0066c958 0066d400
#@category OpenRebellion

listing = currentProgram.getListing()
function_manager = currentProgram.getFunctionManager()
reference_manager = currentProgram.getReferenceManager()

for raw_address in getScriptArgs():
    address = toAddr(raw_address)
    println("references to %s" % address)
    references = reference_manager.getReferencesTo(address)
    found = False
    for reference in references:
        found = True
        source = reference.getFromAddress()
        function = function_manager.getFunctionContaining(source)
        function_label = "<no function>"
        if function is not None:
            function_label = "%s %s" % (function.getName(), function.getEntryPoint())
        instruction = listing.getInstructionAt(source)
        instruction_label = str(instruction) if instruction is not None else "<no instruction>"
        println("  %s %s %s :: %s" %
                (reference.getReferenceType(), source, function_label, instruction_label))
    if not found:
        println("  <none>")

File: ghidra/scripts/DumpReferences.java
// Print references to selected addresses with containing function and instruction.
// Usage with analyzeHeadless:
//   -postScript DumpReferences.java 0066c958 0066d400
//@category OpenRebellion

import ghidra.app.script.GhidraScript;
import ghidra.program.model.listing.Function;
import ghidra.program.model.listing.Instruction;

public class DumpReferences extends GhidraScript {
    @Override
    public void run() throws Exception {
        var listing = currentProgram.getListing();
        var functions = currentProgram.getFunctionManager();
        var references = currentProgram.getReferenceManager();

        for (String rawAddress : getScriptArgs()) {
            var address = toAddr(rawAddress.replace("0x", ""));
            println("references to " + address);
            var iterator = references.getReferencesTo(address);
            boolean found = false;
            while (iterator.hasNext()) {
                monitor.checkCancelled();
                found = true;
                var reference = iterator.next();
                var source = reference.getFromAddress();
                Function function = functions.getFunctionContaining(source);
                String functionLabel = function == null
                    ? "<no function>"
                    : function.getName() + " " + function.getEntryPoint();
                Instruction instruction = listing.getInstructionContaining(source);
                String instructionLabel = instruction == null
                    ? "<no instruction>"
                    : instruction.toString();
                println("  " + reference.getReferenceType() + " " + source + " "
                    + functionLabel + " :: " + instructionLabel);
            }
            if (!found) {
                println("  <none>");
            }
        }
    }
}
